ocaml/xapi/xapi_sr.ml

```diff
@@ -424,22 +424,47 @@ let maybe_copy_sr_rrds ~__context ~sr =
     with Rrd_interface.Rrdd_error(Archive_failed(msg)) ->
       warn "Archiving of SR RRDs to stats VDI failed: %s" msg
 
+(* Removes the SR and dependent objects from the database (VDIs and PBDs).
+   This function is meant to be used only inside this module.
+   Also removes the associated metrics from xcp-rrdd until SR metrics are
+   correctly reported as SR ones (CA-325582)
+ *)
+let _destroy ~__context ~sr =
+  (* CA-325582: Since the pbds are not mounted anymore there are no more
+     metrics being reported, the current ones can be deleted. *)
+  let short_uuid = String.sub (Db.SR.get_uuid ~__context ~self:sr) 0 8 in
+  let is_sr_metric = Astring.String.is_suffix ~affix:short_uuid in
```

Comment:
i would hide `short_uuid` since `is_sr_metric` is the thing you care about

```
let is_sr_metric =
  let short_uuid = String.sub (Db.SR.get_uuid ~__context ~self:sr) 0 8 in
  Astring.String.is_suffix ~affix:short_uuid in
in
```

Reply:
The binding is only available inside the function, I don't see the benefit of hiding it to the rest of the function as it doesn't seem to cause any conflicts or confusing situation and I 'd prefer to have a flat structure if there aren't obvious downsides.